src/datasets/utils/streaming_download_manager.py

```diff
@@ -512,18 +514,36 @@ def iter_archive(self, urlpath: str):
             Generator yielding tuple (path_within_archive, file_obj).
             File-Obj are opened in byte mode (io.BufferedReader)
         """
-        with xopen(urlpath, "rb", use_auth_token=self.download_config.use_auth_token) as f:
-            stream = tarfile.open(fileobj=f, mode="r|*")
-            for tarinfo in stream:
-                file_path = tarinfo.name
-                if not tarinfo.isreg():
-                    continue
-                if file_path is None:
-                    continue
-                if os.path.basename(file_path).startswith(".") or os.path.basename(file_path).startswith("__"):
-                    # skipping hidden files
-                    continue
-                file_obj = stream.extractfile(tarinfo)
-                yield (file_path, file_obj)
-                stream.members = []
-            del stream
+        extension = _get_extraction_protocol(urlpath)
```

Comment:
Currently `_get_extraction_protocol` raises an error for TAR. Therefore you may have to remove the part of `_get_extraction_protocol` that says
```python
    elif path.endswith(".tar.gz") or path.endswith(".tgz"):
        raise NotImplementedError(
            f"Extraction protocol for TAR archives like '{urlpath}' is not implemented in streaming mode. Please use `dl_manager.iter_archive` instead."
        )
```
This verification can be moved in the `_extract` method after calling `_get_extraction_protocol` I think